### PR TITLE
Rewrite AvatarOverlay component with React hooks

### PR DESCRIPTION
**Pull request by takayamaki**
This PR rewrites AvatarOverlay component with React hooks.
And fix animation behavior when component hoverd.

### CODE REVIEW

File: app/javascript/mastodon/components/avatar_overlay.jsx [hunk 1]
```diff
@@ -1,51 +0,0 @@
...
-    size: 46,
-    baseSize: 36,
-    overlaySize: 24,
-  };
-
-  state = {
-    hovering: false,
-  };
-
-  handleMouseEnter = () => {
```
Comment (takayamaki) [inline app/javascript/mastodon/components/avatar_overlay.jsx:29]: Both event handler function object of handleMouseEnter and handleMouseLeave are unused this component.

File: app/javascript/mastodon/components/avatar_overlay.jsx [hunk 2]
```diff
@@ -1,51 +0,0 @@
...
-    this.setState({ hovering: false });
-  };
-
-  render() {
-    const { account, friend, animate, size, baseSize, overlaySize } = this.props;
-    const { hovering } = this.state;
-
-    return (
-      <div className='account__avatar-overlay' style={{ width: size, height: size }}>
-        <div className='account__avatar-overlay-base'><Avatar animate={hovering || animate} account={account} size={baseSize} /></div>
```
Comment (takayamaki) [inline app/javascript/mastodon/components/avatar_overlay.jsx:45]: No need to pass animate arguments  to Avatar component why they have hovering state and its event handler in themself.

File: app/javascript/mastodon/components/avatar_overlay.jsx [hunk 3]
```diff
@@ -1,51 +0,0 @@
...
-import ImmutablePropTypes from 'react-immutable-proptypes';
-import { autoPlayGif } from '../initial_state';
-import Avatar from './avatar';
-
-export default class AvatarOverlay extends React.PureComponent {
-
-  static propTypes = {
-    account: ImmutablePropTypes.map.isRequired,
-    friend: ImmutablePropTypes.map.isRequired,
-    animate: PropTypes.bool,
```
Comment (takayamaki) [inline app/javascript/mastodon/components/avatar_overlay.jsx:12]: animate argument is unused.
https://github.com/mastodon/mastodon/blob/f05fb51ecb34fd5ef86fad861e16cfc959fe0b27/app/javascript/mastodon/components/status.jsx#L505
https://github.com/mastodon/mastodon/blob/f05fb51ecb34fd5ef86fad861e16cfc959fe0b27/app/javascript/mastodon/features/account_timeline/components/moved_note.jsx#L27
https://github.com/mastodon/mastodon/blob/f05fb51ecb34fd5ef86fad861e16cfc959fe0b27/app/javascript/mastodon/components/__tests__/avatar_overlay-test.jsx#L24

File: app/javascript/mastodon/components/avatar_overlay.jsx (hunk 2 above)
Comment (Gargron) [reply app/javascript/mastodon/components/avatar_overlay.jsx:45]: Looks like the event handlers were not hooked up but I think the intent was that both avatars would play at the same time when the overlay was hovered.

Comment (takayamaki) [reply app/javascript/mastodon/components/avatar_overlay.jsx:45]: > Looks like the event handlers were not hooked up but I think the intent was that both avatars would play at the same time when the overlay was hovered.

I seem so too.
But this PR is a refactor, so I didn't change the behavior.
Should I fix it now?

This movie is behavior of current main branch(e5c0b1673).
https://user-images.githubusercontent.com/24884114/232274621-b9cca042-1533-452f-8f65-ee4fd2e504f3.mp4

Comment (takayamaki) [reply app/javascript/mastodon/components/avatar_overlay.jsx:45]: I made a commit.
https://github.com/mastodon/mastodon/pull/24543/commits/d770fd24b6c8f2d4dc6bd031be4b3a0e86c77e6e

https://user-images.githubusercontent.com/24884114/232283584-52e1290b-c44b-44f8-8d6e-544ec1bd5952.mp4

**Review by Gargron — APPROVED** [review]

**Pull request merged**